framework/test/unit/modules/chain/unit/modules/blocks/process.js

```diff
@@ -1716,145 +1723,121 @@ describe('blocks/process', async () => {
 				});
 			});
 
-			describe('when query returns transactions', async () => {
-				beforeEach(() => {
+			describe('when query returns transactions', () => {
+				beforeEach(async () => {
 					modules.transactions.getUnconfirmedTransactionList.returns([
 						{ id: 1, type: 0 },
 						{ id: 2, type: 1 },
 					]);
-					return modules.blocks.verify.processBlock.callsArgWith(
+					modules.blocks.verify.processBlock.callsArgWith(
 						3,
 						null,
 						modules.blocks.verify.processBlock.args
 					);
 				});
 
-				describe('modules.accounts.getAccount', async () => {
-					describe('when fails', async () => {
-						beforeEach(() =>
-							modules.accounts.getAccount.callsArgWith(
-								1,
-								'accounts.getAccount-ERR',
-								null
+				describe('modules.processTransactions.verifyTransactions', () => {
+					describe('when transaction initializations fail', async () => {
+						beforeEach(async () =>
+							modules.processTransactions.verifyTransactions.rejects(
+								new Error('Invalid field types')
 							)
 						);
 
-						it('should call a callback with error', done => {
+						it('should call a callback with error', async () => {
 							blocksProcessModule.generateBlock(
 								{ publicKey: '123abc', privateKey: 'aaa' },
 								41287231,
 								err => {
-									expect(err).to.equal('Sender not found');
-									done();
+									expect(err.message).to.eql('Invalid field types');
 								}
 							);
 						});
 					});
 
-					describe('when succeeds', async () => {
-						beforeEach(() =>
-							modules.accounts.getAccount.callsArgWith(1, null, true)
-						);
-						afterEach(
-							async () =>
-								expect(modules.blocks.verify.processBlock.calledOnce).to.be.true
+					describe('when transactions verification fails', async () => {
+						beforeEach(async () =>
+							modules.processTransactions.verifyTransactions.resolves([
+								{ id: 1, status: 0, errors: [] },
+								{ id: 2, status: 0, errors: [] },
+							])
 						);
 
-						describe('library.logic.transaction.ready', async () => {
-							describe('when returns false', async () => {
-								beforeEach(() =>
-									library.logic.transaction.ready.returns(false)
-								);
-
-								it('should generate block without transactions', done => {
-									blocksProcessModule.generateBlock(
-										{ publicKey: '123abc', privateKey: 'aaa' },
-										41287231,
-										err => {
-											expect(err).to.be.null;
-											expect(library.logic.transaction.verify.calledOnce).to.be
-												.false;
-											expect(
-												modules.blocks.verify.processBlock.args[0][0]
-													.transactions.length
-											).to.equal(0);
-											done();
-										}
-									);
-								});
-							});
-
-							describe('when returns true', async () => {
-								beforeEach(() => library.logic.transaction.ready.returns(true));
+						it('should generate block without transactions', async () => {
```

Comment:
I don't understand how the function can be async and wait for the callback. Can you please double check and maybe change this async function to the one which uses `done` callback.